pandas/plotting/_matplotlib/core.py

```diff
@@ -969,6 +967,10 @@ def _make_plot(self):
         else:
             c_values = c
 
+        # don't plot a colorbar if `c` is a column containing color names
+        c_is_numeric_col = c_is_column and is_numeric_dtype(c_values)
+        cb = self.kwds.pop("colorbar", self.colormap or c_is_numeric_col)
```

Comment:
emm, i meant something like this:
```python
# plot colorbar if 1. colormap is assigned, 2.`c` is a column containing only numeric values
plot_colorbar = self.colormap or c_is_column
cb = self.kwds.pop("colorbar", is_numeric_dtype(c_values) and plot_colorbar)
```
because even when colormap is defined, if not all values of `c` column are numeric, we still dont want to plot colorbar, because it doesnt make sense so far. What do you think?

Reply:
Yes, you're right, thanks! The way I'd originally written it there would still be a colorbar if the user passed something to `cmap`